api/institutions/authentication.py

```diff
@@ -386,4 +388,14 @@ def authenticate(self, request):
                 sso_department=department,
             )
 
+        # Storage region is only updated if user is created via institutional SSO
+        # Storage region is set using the primary institution's settings
+        # The first region in the default region list is used
+        if is_created:
+            user_settings = OSFStorageUserSettings.objects.get(owner=user)
+            default_regions = institution.storage_regions.filter(institutionstorageregion__is_preferred=True)
+            if default_regions and user_settings.default_region not in default_regions:
+                user_settings.default_region = default_regions.first()
+                user_settings.save()
```

Comment:
For CR: In case that multiple preferred regions are added to one institution, use `filter()` and `first()` doesn't break the SSO.